openapi.yaml

```diff
@@ -1,279 +1,662 @@
+              schema:
+                $ref: "#/components/schemas/Errors"
+        "500":
+          description: "Internal Server Error"
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/Errors"
+components:
+  parameters:
+    pretty:
+      name: pretty
+      in: query
+      required: false
+      description: If true, JSON response will be pretty-printed
+      schema:
+        type: boolean
+  securitySchemes:
+    OAuth2:
+      type: oauth2
+      flows: 
+        clientCredentials:
+          tokenUrl: https://api.oregonstate.edu/oauth2/token
+          refreshUrl: https://api.oregonstate.edu/oauth2/token
+          scopes: 
+            full: full access to the API
+  schemas:
+    LocationsResultObject:
+      properties:
+        links:
+          $ref: "#/components/schemas/Links"
+        data:
+          $ref: "#/components/schemas/LocationsResourceObject"
+        included:
+          type: array
+          items:
+            $ref: "#/components/schemas/ServicesResourceObject"
+    LocationsResultObjects:
+      properties:
+        links:
+          $ref: "#/components/schemas/Links"
+        data:
+          type: array
+          items:
+            $ref: "#/components/schemas/LocationsResourceObject"
+        included:
+          type: array
+          items:
+            $ref: "#/components/schemas/ServicesIncluded"
+    ServicesResultObject:
+      properties:
+        links:
+          $ref: "#/components/schemas/Links"
+        data:
+          $ref: "#/components/schemas/ServicesResourceObject"
+    ServicesIncluded:
+      properties:
+        id:
+          type: string
+        type:
+          type: string
+        attributes:
+          $ref: "#/components/schemas/ServicesAttributes"
+        links:
+          $ref: "#/components/schemas/ServiceLinks"
+    LocationsResourceObject:
+      properties:
+        id:
+          type: string
+        type:
+          type: string
+        attributes:
+          $ref: "#/components/schemas/Attributes"
+        links:
+          $ref: "#/components/schemas/SelfLink"
+        relationships:
+          $ref: "#/components/schemas/RelationshipsInLocations"
+    ServicesResultObjects:
+      properties:
+        links:
+          $ref: "#/components/schemas/Links"
+        data:
+          type: array
+          items:
+            $ref: "#/components/schemas/ServicesResourceObject"
+    ServicesResourceObject:
+      properties:
+        id:
+          type: string
+        type:
+          type: string
+        attributes:
+          $ref: "#/components/schemas/ServicesAttributes"
+        links:
+          $ref: "#/components/schemas/ServiceLinks"
+        relationships:
+          $ref: "#/components/schemas/RelationshipsInServices"
+    SelfLink:
+      properties:
+        self:
+          type: string
+          description: Url to this resource / location.
+    Attributes:
```

Comment:
Can you add example, description, and format/pattern (if applied) to all of these attributes?